### ISSUE TITLE
`bleach` unconstrained dependency causes import errors

### ISSUE THREAD

**Issue by ReubenFrankel** [jupyter/nbconvert#1754]
The [`bleach`](https://pypi.org/project/bleach/) package was updated to version `5.0.0` earlier today. This broke importing modules from `nbconvert` on the latest version (`6.4.5`). This seems to be an issue with older `6.4` versions as well.

The `bleach` dependency is currently unconstrained: https://github.com/jupyter/nbconvert/blob/68b496b7fcf4cfbffe9e1656ac52400a24cacc45/setup.py#L248

We tested with the previous major version of `bleach` (`4.1.0`), which resolved the issue.

---

### Steps to reproduce

(tested with Python 3.7.13 and 3.9.5)

```
pip install nbconvert
```

```py
from nbconvert import MarkdownExporter
```

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/private/tmp/.venv/lib/python3.7/site-packages/nbconvert/__init__.py", line 4, in <module>
    from .exporters import *
  File "/private/tmp/.venv/lib/python3.7/site-packages/nbconvert/exporters/__init__.py", line 3, in <module>
    from .html import HTMLExporter
  File "/private/tmp/.venv/lib/python3.7/site-packages/nbconvert/exporters/html.py", line 26, in <module>
    from nbconvert.filters.highlight import Highlight2HTML
  File "/private/tmp/.venv/lib/python3.7/site-packages/nbconvert/filters/__init__.py", line 6, in <module>
    from .markdown import *
  File "/private/tmp/.venv/lib/python3.7/site-packages/nbconvert/filters/markdown.py", line 23, in <module>
    from .pandoc import convert_pandoc
  File "/private/tmp/.venv/lib/python3.7/site-packages/nbconvert/filters/pandoc.py", line 1, in <module>
    from nbconvert.utils.pandoc import pandoc
  File "/private/tmp/.venv/lib/python3.7/site-packages/nbconvert/utils/pandoc.py", line 12, in <module>
    from nbconvert.utils.version import check_version
  File "/private/tmp/.venv/lib/python3.7/site-packages/nbconvert/utils/version.py", line 11, in <module>
    from packaging.version import Version
ModuleNotFoundError: No module named 'packaging'
```

**Comment by bollwyvl** [comment]
ideally, we'd be able to add a dependency on `packaging`, which #1755 is exploring. it's already a dependency of many downstreams.

**Comment by bollwyvl** [comment]
update: #1755 offers one approach to move forward. we'll see if others want to do alternative PRs.

**Comment by blink1073** [comment]
Fixed by #1755
